src/state/slices/opportunitiesSlice/selectors.ts

```diff
@@ -115,50 +155,302 @@ export const selectUserStakingOpportunitiesByStakingId = createDeepEqualOutputSe
+      ...userStakingOpportunity,
+      chainId: fromAssetId(userStakingOpportunity.assetId ?? '').chainId,
+      rewardsAmountCryptoPrecision: userStakingOpportunity.rewardsAmountCryptoPrecision ?? '',
+    }
+  },
+)
+
+// The same as the previous selector, but parsed as an EarnOpportunityType
+// TODO: testme
+export const selectAggregatedEarnUserLpOpportunity = createDeepEqualOutputSelector(
+  selectLpOpportunitiesById,
+  selectLpIdParamFromFilter,
+  selectPortfolioCryptoHumanBalanceByFilter,
+  selectAssets,
+  selectMarketData,
+  (
+    lpOpportunitiesById,
+    lpId,
+    aggregatedLpAssetBalance,
+    assets,
+    marketData,
+  ): FoxFarmingEarnOpportunityType | undefined => {
+    if (!lpId || !aggregatedLpAssetBalance) return
+
+    const marketDataPrice = marketData[lpId as AssetId]?.price
+    const opportunityMetadata = lpOpportunitiesById[lpId]
+    const baseLpEarnOpportunity = LP_EARN_OPPORTUNITIES[lpId]
+
+    if (!opportunityMetadata || !baseLpEarnOpportunity) return
+
+    const [underlyingToken0Amount, underlyingToken1Amount] =
+      opportunityMetadata.underlyingAssetIds.map((assetId, i) =>
+        bnOrZero(aggregatedLpAssetBalance)
+          .times(
+            fromBaseUnit(
+              opportunityMetadata?.underlyingAssetRatios[i] ?? '0',
+              assets[assetId].precision,
+            ),
+          )
+          .toFixed(6)
           .toString(),
-      }
-    }, undefined)
+      )
+
+    const opportunity = {
+      ...baseLpEarnOpportunity,
+      ...opportunityMetadata,
+      isLoaded: true,
+      chainId: fromAssetId(lpId as AssetId).chainId,
+      underlyingToken1Amount,
+      underlyingToken0Amount,
+      cryptoAmount: aggregatedLpAssetBalance,
+      fiatAmount: bnOrZero(aggregatedLpAssetBalance)
+        .times(marketDataPrice ?? '0')
+        .toString(),
+      icons: opportunityMetadata.underlyingAssetIds.map(assetId => assets[assetId].icon),
+    }
+
+    return opportunity
   },
 )
 
+// "Give me the total values over all my accounts aggregated into one for each opportunity, and then aggregated these into one final value"
+// TODO: testme
+export const selectAggregatedUserStakingOpportunity = createDeepEqualOutputSelector(
+  selectAggregatedUserStakingOpportunities,
+  (aggregatedOpportunities): UserStakingOpportunity & OpportunityMetadata =>
+    aggregatedOpportunities.reduce<UserStakingOpportunity & OpportunityMetadata>(
+      (acc, currentOpportunity) => {
+        return {
+          ...acc,
+          ...currentOpportunity,
+          stakedAmountCryptoPrecision: bnOrZero(currentOpportunity.stakedAmountCryptoPrecision)
+            .plus(acc.stakedAmountCryptoPrecision)
+            .toString(),
+          rewardsAmountCryptoPrecision: bnOrZero(currentOpportunity.rewardsAmountCryptoPrecision)
+            .plus(acc.rewardsAmountCryptoPrecision)
+            .toString(),
+        }
+      },
+      {} as UserStakingOpportunity & OpportunityMetadata,
```

Comment:
can prob remove this cast?

Reply:
✅  b95e74e2d09c5f926c52796fd49b7d52e3a705ed